pkg/shell/cockpit-server.js

```diff
@@ -868,6 +868,13 @@ PageShutdownDialog.prototype = {
                 shell.show_unexpected_error(ex);
             })
             .done(function(ex) {
+                var expect_delay = parseInt(delay);
```

Comment:
Hmmm, not sure about the whole concept of delays. These could be days away. If this is the only use case of the concept of delays in expect restart, then I would skip on it for now.

For this particular case we need to listen to the logind Dbus API and get our restart information from there. Once we're at around 60 seconds, just tell the outer frame that a restart is expected and leave it at that, no?

Again, such work should be a follow up. I'm just leary of cementing the concept of delays into the API.